src/js/trove/parse-pyret.js

```diff
@@ -1344,31 +1345,56 @@
         if (countParses === 1) {
           var ast = grammar.constructUniqueParse(parsed);
           //          console.log(ast.toString());
-          return translate(ast, fileName);
+          return RUNTIME.ffi.makeRight(translate(ast, fileName));
         } else {
           var asts = grammar.constructAllParses(parsed);
           throw "Non-unique parse";
           for (var i = 0; i < asts.length; i++) {
             //console.log("Parse " + i + ": " + asts[i].toString());
             //            console.log(("" + asts[i]) === ("" + asts2[i]));
           }
-          return translate(ast, fileName);
+          return RUNTIME.ffi.makeRight(translate(ast, fileName));
         }
       } catch(e) {
-        // console.error("Fatal error in parsing: ", e);
-        throw e;
+        if (RUNTIME.isPyretException(e)) {
+          return RUNTIME.ffi.makeLeft(RUNTIME.makeObject({
```

Comment:
This feels a bit odd to me -- having this not-quite-an-exception object that colludes with the handler code below just seems wrong, and more than a little bit non-type-correct.  @jpolitz is there a "cleaner" or "more officially sanctioned" design you can think of, that might tie in better with the existing compile- and runtime errors we have?  I feel like there should be, but I can't think of it yet.